### ISSUE TITLE
Group selection of large numbers of individual files

### ISSUE THREAD

**Issue by roofrack9** [ganeshrvel/openmtp#243]
Good morning,
I am using your OpenMTP and I made a donation and requested a group select option and you suggested I open an issue here.  

By this I mean I was hoping it would emulate the behaviour of Mac OS Finder in that, when selecting files, if I want to select multiple files I can click the top one, then shift+click the bottom one and all files in between will be selected.  However from what I see currently with OpenMTP if you want to select and move a large group of files you have to select them all individually.  I may be wrong but I cannot find a way to select multiple files at a time without clicking on every single one.  
It would be a very useful addition if it is easy to implement.  
Regards

**Comment by resuna** [comment]
Me too! There's a standard for multi-select on the Mac and it'd be great if it was followed.

**Comment by alecmuffett** [comment]
+1

**Comment by bkmystim** [comment]
awesome software! but yes - my main missing thing is group/multiple selection of files.
if I want to copy 200 pictures, I have to select every one of them.
multiple file selection would be a major enhancement!

**Comment by bierlaster** [comment]
+1

**Comment by ganeshrvel** [comment]
There is a check box available in the list view. Could you see if that works for you?

**Comment by bierlaster** [comment]
> There is a check box available in the list view. Could you see if that works for you?
Hi @ganeshrvel, it works, but not for my use case: I have a folder of 500 pictures and want to sync the latest 100. So either I manually select 100 files or select all and manually unselect 400 files...

**Comment by ganeshrvel** [comment]
Folks, i know there are a few backlogs here and lots of feature requests. I have a lot of things happening back at home and work.

Give me a few weeks so that I can get back into completing these features for you.

Cheers

**Comment by chaehni** [comment]
Thanks. I appreciate it a lot!
This app is a real life saver.

**Comment by gagomes** [comment]
@ganeshrvel  Any updates? Thanks :-)

**Comment by bednarikjan** [comment]
+1 for this feature, it would make life much easier

Otherwise, awesome piece of software!

**Comment by UnnamedHero** [comment]
+1

**Comment by alecmuffett** [comment]
This feature already exists but it is rather unintuitive.

I wrote up my experience at https://alecmuffett.com/article/15274

**Comment by lightheaded** [comment]
I don't think this works on the MacOS app. Otherwise, the app is absolutely brilliant, but I have to return to Android File Transfer to reasonably copy a few days' worth of videos off my phone. In any case, thanks for spending your time on this free tool ♥️

**Comment by resuna** [comment]
Alec, the problem is that these options still require individually selecting every file.

**Comment by gsouvik** [comment]
> This feature already exists but it is rather unintuitive. I wrote up my experience at https://alecmuffett.com/article/15274

Thanks for pointing this out. This doesn't work for me on list view. Also on grid view the left and right arrow clicks only work, the top and bottom doesn't.  As many others mentioned, this is a brilliant app, save this one shortcoming.

**Comment by jonahclarsen** [comment]
Yeah group select is way too laggy to realistically use. It would be way better if up/down worked on grid view, but ideally selecting one thing then shift+clicking another thing should select all things in between them, like it does in Finder and File Explorer.

**Comment by TomWinder** [comment]
+1

**Comment by ganeshrvel** [comment]
I'm planning a lightweight and modern rewrite of the app future. It's already in progress. This will be a complete GUI app overhaul from the ground up. I will try to add this feature into it.

**Comment by iforwms** [comment]
+1

**Comment by z-l-p** [comment]
+1 for me too. Batch-select would make this a killer app!

**Comment by danielo515** [comment]
Hey, thank you for this great piece of software.
I'm in the same situation, where I want to select around 200 files out of 300. Will be great to have this feature

**Comment by epictutorials** [comment]
Hi. Love this software, but the inability to shift-select multiple files really kills its utility. I want to use it to import hundreds (occasionally thousands) of photos/videos from Android device to Mac, but can't. I will gladly pay for this functionality.

**Comment by hunterirving** [comment]
Big fan of the app - just wanted to say that I'd love to see this feature too! Thanks so much for creating this, and thanks again for making it open source!

**Comment by RobMHKaw** [comment]
Great app, love it. Works smoothly but I would like to add the request as well to be able to select multiple files with the shift key. Hope there will be an update soon. Cheers

**Comment by teqtic** [comment]
Please add this! Even a drag selection would be great!

**Comment by lgespee** [comment]
The multi select doesn't behave as expected per se, one would expect clicking a file (first of 100) then `shift + click` on the 100th file, that all the 98 files in between are also selected. Instead only files 1 and 100 are selected.

Best next option (which saves a lot of clicks) is selecting a single file in list view, and then using `shift + down` or `shift + up`.
Keep holding both buttons to easily select a longer list of files.

The GitHub [repo's main page](https://github.com/ganeshrvel/openmtp) shows the README.md file which says:
> Select Multiple Items Forward (in List View) | shift+up
> Select Multiple Items Backward (in List View) | shift+down

Only downside is the selection isn't maintained, so after copying the files over you can't see which files you selected and delete them, you have to select them once more (and make sure it's the exact same ones).

**Comment by ataetgi** [comment]
++

**Comment by ralph303** [comment]
Are there any news on that topic? I use and like OpenMTP, but moving around files is really uncomfortable and a multiselection by pressing Shift (just like in Finder or Windows-Explorer) would be really great.

**Comment by ganeshrvel** [comment]
New app in progress. You should have it in the nextgen openmtp

**Comment by ganeshrvel** [comment]
Currently, you can navigate or multi-select files like this in OpenMTP:

https://github.com/user-attachments/assets/84935b8a-5868-48d2-bf0d-42b3ed33fcf8

<img width="533" height="623" alt="Image" src="https://github.com/user-attachments/assets/3c06b06b-3385-484a-84df-e6b0ca79ef54" />

Range selection will be implemented in the next-gen OpenMTP.

**Comment by teqtic** [comment]
> New app in progress. You should have it in the nextgen openmtp

Do you have a new repository for the new app? Why not update this one?

**Comment by ralph303** [comment]
To be clear, the most important range selection, at least when handling lots of files, is this one:
- left click file number 1
- shift+left click file number 300 (for example)
=> all files between those two are automatically selected / highlighted!

Until then I use the workaround to copy whole folders and delete all unneeded files in Finder, because shift-select is too slow for large numbers of files.
It's great, that this feature is planned for the new version!

**Comment by ganeshrvel** [comment]
Range selection using Click, Cmd+Click, Shift+Click, and Shift+Arrow navigation was internally achieved. The behavior closely mirrors native macOS Table View selection. NextGen OpenMTP is coming along nicely and getting closer to its initial alpha release.
